### PR TITLE
CgmAbsSurvey stuff

### PR DESCRIPTION
**Pull request by jnburchett**
Functions to build CGMAbsSurvey objects; also sticks the impact parameter in returned ion_tbls

### CODE REVIEW

File: pyigm/cgm/utils.py [hunk 1]
```diff
@@ -84,6 +84,99 @@ def calc_rho(galaxy, igm_sys, cosmo, ang_sep=None, correct_lowz=True,
     return rho, ang_sep
 
 
+def get_close_galaxies(field,rholim=300.*u.kpc,minz=0.001,maxz=None):
+    '''
+    Generate table of galaxies close to sightlines for one more IgmGalaxyField
```
Comment (profxj) [inline pyigm/cgm/utils.py]: one *or* more  ?

File: pyigm/cgm/utils.py [hunk 2]
```diff
@@ -84,6 +84,99 @@ def calc_rho(galaxy, igm_sys, cosmo, ang_sep=None, correct_lowz=True,
...
+
+    Parameters
+    ----------
+    fields : IgmGalaxyField or list of IgmGalaxyField
+      Fields to be included
+    rholim : quantity, optional
+      Maximum impact parameter limit of galaxies
+    minz : float, optional
```
Comment (profxj) [inline pyigm/cgm/utils.py:97]: Define maxz

File: pyigm/cgm/utils.py [hunk 3]
```diff
@@ -84,6 +84,99 @@ def calc_rho(galaxy, igm_sys, cosmo, ang_sep=None, correct_lowz=True,
...
+
+    Returns
+    -------
+    closegaltab : astropy Table
+      Table of galaxies meeting impact parameter and redshift criteria
+    '''
+    if maxz is None:
+        maxz = np.inf
+    field.galaxies['rho'] = field.calc_rhoimpact(field.galaxies,comoving=False)
+    idxs=np.where((field.galaxies['rho'].quantity<rholim)&
```
Comment (profxj) [inline pyigm/cgm/utils.py]: Just use logic here, i.e. no np.where

e.g.

```
close = field.galaxies['rho'] < rholim
goodz = (field.galaxies['Z'] > minz) & (field.galaxies['Z'] < maxz)
good_gals = close & goodz
```

File: pyigm/cgm/utils.py [hunk 4]
```diff
@@ -84,6 +84,99 @@ def calc_rho(galaxy, igm_sys, cosmo, ang_sep=None, correct_lowz=True,
...
+    if ((not isinstance(fields,list))|(not isinstance(sightlines,list))|
+        (len(fields) != len(sightlines))):
+        raise IOError("Inputs fields and sightlines must lists of the same length")
+
+    from pyigm.cgm.cgmsurvey import CGMAbsSurvey
+    cgmsys = []
+    for i,ff in enumerate(fields):
+        thiscgmlist = cgmabssys_from_sightline_field(ff,sightlines[i],**kwargs)
+        cgmsys.extend(thiscgmlist)
+    if name is not None:
```
Comment (profxj) [inline pyigm/cgm/utils.py:175]: This seems a bit strange.  The way we instantiate
depends on whether the name variable is set?
Check this.

**Review by profxj — APPROVED** [review]
Just a series of minor things for you to consider before merging.

Any chance you can include a test or two?

**Comment by profxj** [comment]
Merge if happy

File: pyigm/cgm/utils.py (hunk 1 above)
Comment (jnburchett) [reply pyigm/cgm/utils.py]: typo -- should be '...for an IgmGalaxyField...'

File: pyigm/cgm/utils.py (hunk 2 above)
Comment (jnburchett) [reply pyigm/cgm/utils.py:97]: good catch

File: pyigm/cgm/utils.py (hunk 3 above)
Comment (jnburchett) [reply pyigm/cgm/utils.py]: yep

File: pyigm/cgm/utils.py (hunk 4 above)
Comment (jnburchett) [reply pyigm/cgm/utils.py:175]: Ah, the 'else' case should also instantiate as CGMAbsSurvey.from_cgmabssys() but without declaring a 'survey' argument.

**Pull request merged**